servicetest/filesystem/test_filesystem.py

```diff
@@ -251,29 +262,159 @@ def test_tmpfs_writebuffer_size_enabled(self):
+            make sure that SoftwareContainerAgent behaves as expected when
+            the bad configuration is sent to it.
+
+            This tests that the container is created even if the size parameter
+            is missing from the configuration.
+        """
+        ca = Container()
+
+        absoluteTestFile = None
+
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                               "enableTemporaryFileSystemWriteBuffer": true}]'
 
         try:
-            ca.start(DATA)
+            id = ca.start(DATA)
+
+            absoluteTestFile = os.path.join("/tmp/container/SC-" + str(id) +
+                                            "/",
+                                            TESTFILE)
+
+            with open(absoluteTestFile, "w") as f:
+                f.write("w" * 95 * 1024 * 1024)
+
+            # Make sure the file is between 94 and 96 megabyte (exact size
+            # may vary depending on platform, blocksize etc)
+            assert os.path.getsize(absoluteTestFile) >= 94 * 1024 * 1024
+            assert os.path.getsize(absoluteTestFile) <= 96 * 1024 * 1024
 
-            absoluteTestFile = os.path.join("/usr/var/lib/lxc/SC-0/rootfs",
+            # Try to write a file that is too large for the filesystem (105
+            # megabytes). This should fail with an IOError.
+            with open(absoluteTestFile, "w") as f:
+                with pytest.raises(IOError):
+                    f.write("w" * 105 * 1024 * 1024)
+        finally:
+            ca.terminate()
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_tmpfs_writebuffer_size_missing_tmpfs_enable_flag(self):
+        """ This test will test various bad configuration combinations and
+            make sure that SoftwareContainerAgent behaves as expected when
+            the bad configuration is sent to it.
+
+            This run should work, but no temporary filesystem will be setup as
+            the enableTemporaryFileSystemWriteBuffer defaults to false and no
+            parsing will be performed on the temporaryFileSystemSize parameter.
+        """
+        ca = Container()
+
+        absoluteTestFile = None
+
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                               "temporaryFileSystemSize": ' + str(TenMB) + '}]'
+
+        try:
+            id = ca.start(DATA)
+
+            absoluteTestFile = os.path.join("/tmp/container/SC-" + str(id) +
+                                            "/",
+                                            TESTFILE)
+
+            with open(absoluteTestFile, "w") as f:
+                f.write("w" * 95 * 1024 * 1024)
+
+            # Make sure the file is between 94 and 96 megabyte (exact size
+            # may vary depending on platform, blocksize etc)
+            assert os.path.getsize(absoluteTestFile) >= 94 * 1024 * 1024
+            assert os.path.getsize(absoluteTestFile) <= 96 * 1024 * 1024
```

Comment:
The comments above where added to alleviate the understanding of the values. I hope those are enough?